core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java

```diff
@@ -1091,6 +1091,89 @@ public void testDetachAttachDifferentPartitionTableReaderReload() throws Excepti
+                            "2022-10-17T00:00:17.279900Z\t2022-10-18T23:59:59.000000Z\t10000\n");
+
+            compile("ALTER TABLE " + tableName + " DETACH PARTITION LIST '" + partitionName + "'", sqlExecutionContext);
+
+            assertSql("SELECT min(ts), max(ts), count() FROM " + tableName,
+                    "min\tmax\tcount\n" +
+                            "2022-10-18T00:00:16.779900Z\t2022-10-18T23:59:59.000000Z\t5000\n");
+
+            // different location, another volume potentially
+            final CharSequence s3Buckets = temp.newFolder("S3").getAbsolutePath();
+
+            // copy .detached folder to the different location
+            final String detachedPartitionName = partitionName + TableUtils.DETACHED_DIR_MARKER;
+            copyPartitionAndMetadata(
+                    configuration.getRoot(),
+                    tableName,
+                    detachedPartitionName,
+                    s3Buckets,
+                    tableName,
+                    detachedPartitionName,
+                    null
+            );
+
+            // then remove the original .detached folder
+            path.of(configuration.getRoot())
+                    .concat(tableName)
+                    .concat(detachedPartitionName)
+                    .$();
+            Assert.assertEquals(0, Files.rmdir(path));
+            Assert.assertFalse(ff.exists(path));
+
+            // create the .attachable link in the table's data folder
+            // with target the .detached folder in the different location
+            other.of(s3Buckets) // <-- the copy of the now lost .detached folder
+                    .concat(tableName)
+                    .concat(detachedPartitionName)
+                    .$();
+            path.of(configuration.getRoot()) // <-- soft link path
+                    .concat(tableName)
+                    .concat(partitionName)
+                    .put(configuration.getAttachPartitionSuffix())
+                    .$();
+            Assert.assertEquals(0, ff.softLink(other, path));
+
+            // attach the partition via soft link
+            compile("ALTER TABLE " + tableName + " ATTACH PARTITION LIST '" + partitionName + "'", sqlExecutionContext);
+
+            assertSql("SELECT min(ts), max(ts), count() FROM " + tableName,
+                    "min\tmax\tcount\n" +
+                            "2022-10-17T00:00:17.279900Z\t2022-10-18T23:59:59.000000Z\t10000\n");
+
+            assertSql("SELECT * FROM " + tableName + " WHERE ts in '" + partitionName + "' LIMIT -5",
+                    "l\ti\tts\n" +
+                            "4996\t4996\t2022-10-17T23:58:50.380400Z\n" +
+                            "4997\t4997\t2022-10-17T23:59:07.660300Z\n" +
+                            "4998\t4998\t2022-10-17T23:59:24.940200Z\n" +
+                            "4999\t4999\t2022-10-17T23:59:42.220100Z\n" +
+                            "5000\t5000\t2022-10-17T23:59:59.500000Z\n"
+            );
```

Comment:
Would be good to insert into 2022-10-17 at the end and see if the data will be appended to the sym linked partition.

Reply:
And then it will be good to insert to the beginning of `2022-10-17` partition and see if the partition will be copied to the table folder and how the old one will be deleted e.g.

Soft link deleted and files at S3 folder left
Files in S3 folders are deleted

Reply:
Also you need to test reopening `TableWriter` and how it scans partition folders on creation when one of the partitions is a soft link

Reply:
`O3PartitionPurgeJob` has to be tested when one of the partitions is a soft link too.

Reply:
UPDATE needs testing over soft linked partition.

Reply:
Please also test DETACH and DROP partition when the partition is a soft link

Reply:
okis, on it

Reply:
question, when a soft link is provided, the only thing that changes is the location of the partition, all other considerations remain the same, no?, so new partition versions would need to exist in the S3 bucket or local to the table?, Attaching a partition this way is for mostly reads no?